src/webdrivermanager/base.py

```diff
@@ -177,26 +173,32 @@ def _get_latest_version_with_github_page_fallback(self, url, fallback_url, requi
 
         return version  # noqa: R504
 
+    def get_mac_cpu_type(self):
+        # Identify mac CPU type, refer to https://stackoverflow.com/questions/65970469/what-does-platform-system-and-platform-architecture-return-on-apple-m1-silic
+        return "arm" if platform.processor() == "arm" else "intel" if self.os_name == "mac" else ""
+
     def _parse_github_api_response(self, version, response):
-        filenames = [asset["name"] for asset in response.json()["assets"]]
-        filename = [name for name in filenames if self.os_name in name]
-        mac_cpu_type = self.get_mac_cpu_type()
-        
-        if not filename:
-            raise_runtime_error(f"Error, unable to find a download for os: {self.os_name}")
-
-        if len(filename) > 1:
-            if self.os_name is "mac":
-                filename = [name for name in filenames if "aarch64" in name] if mac_cpu_type is "arm" else [name for name in filenames if "aarch64" not in name]
+        all_filenames = [asset['name'] for asset in response.json()['assets']]
+        os_filenames = [name for name in all_filenames if self.os_name in name]
+
+        if not os_filenames:
+            raise_runtime_error('Error, unable to find a download for os: {0}'.format(self.os_name))
+
+        if len(os_filenames) > 1:
+            if self.os_name == "mac":
+                mac_cpu_type = self.get_mac_cpu_type()
+                arch_filenames = [name for name in os_filenames if "aarch64" in name] if mac_cpu_type == "arm" else [name for name in os_filenames if "aarch64" not in name]
```

Comment:
Should we have the same check `if len(arch_filenames) != 1:` for mac?
Something like:
`raise_runtime_error(f"Error, unable to determine correct filename for {self.os_name} ({mac_cpu_type})")`

Reply:
I was thinking that, the pattern has been set to fail fast on multiple results rather than picking the first element in the array.